### PR TITLE
Provide original sequence names for all BGM and rename some sequences

### PR DESCRIPTION
**Pull request by tom-overton**
I was looking at sound assets in OoT3D and MM3D recently, and I realized that the comments on the sequences was missing a lot of names. I took the opportunity to put the MM3D name for every single one in there, and I also did a good bit of clean-up, clarification, and correction on the comments. Lastly, I renamed three of the sequences, which is probably somewhat controversial, but I hope my choices were fine.

### CODE REVIEW

File: src/audio/code_8019AF00.c [hunk 1]
```diff
@@ -330,7 +330,7 @@ u8 sSeqFlags[] = {
     SEQ_FLAG_ENEMY,                    // NA_BGM_CAVERN
     SEQ_FLAG_RESUME | SEQ_FLAG_ENEMY,  // NA_BGM_MILK_BAR
     SEQ_FLAG_FANFARE,                  // NA_BGM_ZELDA_APPEAR
-    0,                                 // NA_BGM_SARIAS_SONG
+    0,                                 // NA_BGM_WOODS_OF_MYSTERY
```
Comment (engineer124) [inline src/audio/code_8019AF00.c:333]: This rename is tricky because I've thought of naming it this before, as well as `NA_BGM_LOST_WOODS` but it is just an identical track to OoT's saria's song, so it feels weird to give it a different name. You can even see it in the name `MAYOIMORI`.

The reason I went with `SARIAS_SONG` over `LOST_WOODS` was that `LOST_WOODS` is a different region in MM so using that name here would be confusing, even though the OG name `MAYOIMORI` translates to lost woods.

Note that there's a similar issue with `NA_BGM_ROMANI_RANCH` which is really just `NA_BGM_LON_LON` (`RONRON`) so there's already a degree of inconsistency with this

File: src/audio/code_8019AF00.c [hunk 2]
```diff
@@ -350,7 +350,7 @@ u8 sSeqFlags[] = {
     SEQ_FLAG_FANFARE,                  // NA_BGM_OATH_TO_ORDER
     0,                                 // NA_BGM_SWORD_TRAINING_HALL
     SEQ_FLAG_FANFARE,                  // NA_BGM_GORON_LULLABY_INTRO
-    SEQ_FLAG_FANFARE,                  // NA_BGM_OCA_FAIRY
+    SEQ_FLAG_FANFARE,                  // NA_BGM_LEARNED_NEW_SONG
```
Comment (engineer124) [inline src/audio/code_8019AF00.c:353]: This rename is fine, but then I wonder what the other `NA_BGM_OCA_` names should be since those are after you play the song, not necessarily when you learn them. The terminology needs clearing up for ocarina anyway which needs to be done with message/audio in sync, so I don't think there's anything to do about it now. So this change is fine.

**Review by engineer124 — COMMENTED** [review]
Thanks for this!

File: src/audio/code_8019AF00.c (hunk 1 above)
Comment (isghj5) [reply src/audio/code_8019AF00.c:333]: Yes but `Woods of mystery` is not `Lost woods`, the actual place this music plays, the maze in southern swamp where we go chase the monkey to find the witch, is called Woods of mystery, so it kinda makes sense to name it after the area instead of the non-existent character.
(although either is fine with me)

File: include/sequence.h [hunk 3]
```diff
@@ -3,132 +3,131 @@
...
+#define NA_BGM_SWAMP_CRUISE 0x0E            // Swamp Cruise ("JUNGLE_CREWS")
+#define NA_BGM_SHARPS_CURSE 0x0F            // Sharp's Curse ("SHARP")
+#define NA_BGM_GREAT_BAY_REGION 0x10        // Great Bay Coast ("SEA_AREA")
+#define NA_BGM_IKANA_REGION 0x11            // Ikana Canyon ("IKANA_AREA")
+#define NA_BGM_DEKU_PALACE 0x12             // Deku Palace ("DEKU_KING")
+#define NA_BGM_MOUNTAIN_REGION 0x13         // Mountain Village ("SNOW_AREA")
+#define NA_BGM_PIRATES_FORTRESS 0x14        // Pirates' Fortress ("PIRATE_AREA")
+#define NA_BGM_CLOCK_TOWN_DAY_1 0x15        // Clock Town Day 1. This song's filename is dummied out in MM3D ("DUMMY_21")
+#define NA_BGM_CLOCK_TOWN_DAY_2 0x16        // Clock Town Day 2. This song's filename is dummied out in MM3D ("DUMMY_22")
+#define NA_BGM_CLOCK_TOWN_DAY_3 0x17        // Clock Town Day 3. This song's filename is dummied out in MM3D ("DUMMY_23")
```
Comment (isghj5) [inline include/sequence.h:27]: These are just pointers in the audioseq index data, which is why they are named dummy, feels weird to document the name being dummied and not that they are pointers to another song slot and list that name here instead.

File: include/sequence.h [hunk 4]
```diff
@@ -3,132 +3,131 @@
...
+#define NA_BGM_CHASE 0x03                   // Forest Chase ("CHASE")
+#define NA_BGM_MAJORAS_THEME 0x04           // Majora's Theme ("STALKID")
+#define NA_BGM_CLOCK_TOWER 0x05             // The Clock Tower ("CLOCK_TOWER")
+#define NA_BGM_STONE_TOWER_TEMPLE 0x06      // Stone Tower Temple ("RUIN_DUNGEON_FRONT")
+#define NA_BGM_INV_STONE_TOWER_TEMPLE 0x07  // Stone Tower Temple Inverted ("RUIN_DUNGEON_REVERSE")
+#define NA_BGM_FAILURE_0 0x08               // Missed Event 1 ("FAILURE_0")
+#define NA_BGM_FAILURE_1 0x09               // Failed Milk Run ("FAILURE_1")
+#define NA_BGM_HAPPY_MASK_SALESMAN 0x0A     // Happy Mask Salesman ("OMENYA")
+#define NA_BGM_SONG_OF_HEALING 0x0B         // Song of Healing ("DEMO_CURE")
+#define NA_BGM_SWAMP_REGION 0x0C            // Southern Swamp and Woodfall ("DAMP_AREA")
```
Comment (isghj5) [inline include/sequence.h:16]: I probably would have named this after the name of the actual region instead of "swamp region", like, the pirate song didn't get named "Pirate_region"
but that's just my opinion not a big deal to me.

File: include/sequence.h [hunk 5]
```diff
@@ -3,132 +3,131 @@
...
+#define NA_BGM_ALIEN_INVASION 0x0D          // Ghost Attack ("UFO")
+#define NA_BGM_SWAMP_CRUISE 0x0E            // Swamp Cruise ("JUNGLE_CREWS")
+#define NA_BGM_SHARPS_CURSE 0x0F            // Sharp's Curse ("SHARP")
+#define NA_BGM_GREAT_BAY_REGION 0x10        // Great Bay Coast ("SEA_AREA")
+#define NA_BGM_IKANA_REGION 0x11            // Ikana Canyon ("IKANA_AREA")
+#define NA_BGM_DEKU_PALACE 0x12             // Deku Palace ("DEKU_KING")
+#define NA_BGM_MOUNTAIN_REGION 0x13         // Mountain Village ("SNOW_AREA")
```
Comment (isghj5) [inline include/sequence.h:23]: This BGM plays through the whole region, specifying "mountain village" in the comment feels too specific.

File: include/sequence.h [hunk 6]
```diff
@@ -3,132 +3,131 @@
...
+#define NA_BGM_PIRATES_FORTRESS 0x14        // Pirates' Fortress ("PIRATE_AREA")
+#define NA_BGM_CLOCK_TOWN_DAY_1 0x15        // Clock Town Day 1. This song's filename is dummied out in MM3D ("DUMMY_21")
+#define NA_BGM_CLOCK_TOWN_DAY_2 0x16        // Clock Town Day 2. This song's filename is dummied out in MM3D ("DUMMY_22")
+#define NA_BGM_CLOCK_TOWN_DAY_3 0x17        // Clock Town Day 3. This song's filename is dummied out in MM3D ("DUMMY_23")
+#define NA_BGM_FILE_SELECT 0x18             // File Select ("FILE_SELECT")
+#define NA_BGM_CLEAR_EVENT 0x19             // Event Clear ("EVENT_CLEAR")
+#define NA_BGM_ENEMY 0x1A                   // Battle ("ENEMY")
+#define NA_BGM_BOSS 0x1B                    // Boss Battle ("BOSS00")
```
Comment (isghj5) [inline include/sequence.h:31]:
```suggestion
#define NA_BGM_BOSS 0x1B                    // Main Dungeon Boss Battle ("BOSS00")
```
As this doesn't play for minibosses or the final boss

File: include/sequence.h [hunk 7]
```diff
@@ -3,132 +3,131 @@
...
+#define NA_BGM_MARINE_RESEARCH_LAB 0x2C     // Marine Research Laboratory ("DRUGSTORE")
+#define NA_BGM_GIANTS_THEME 0x2D            // The Four Giants ("KYOJIN_ROOM")
+#define NA_BGM_SONG_OF_STORMS 0x2E          // Guru-Guru's Song ("FUSHA")
+#define NA_BGM_ROMANI_RANCH 0x2F            // Romani Ranch ("RONRON")
+#define NA_BGM_GORON_VILLAGE 0x30           // Goron Village ("GORON")
+#define NA_BGM_MAYORS_OFFICE 0x31           // Mayor Dotour ("MEETING")
+#define NA_BGM_OCA_EPONA 0x32               // Ocarina Epona's Song ("OCA_EPONA")
+#define NA_BGM_OCA_SUNS 0x33                // Ocarina Sun's Song ("OCA_SUNS")
+#define NA_BGM_OCA_TIME 0x34                // Ocarina Song of Time ("TIME_NORMAL")
+#define NA_BGM_OCA_STORM 0x35               // Ocarina Song of Storms ("OCA_STORM")
```
Comment (isghj5) [inline include/sequence.h:57]: I kinda want OCA to be full length OCARINA, we aren't saving that much but for some reason this shorthand takes me a good 5 seconds to remember OCA is short for ocarina, is that just me tho?

File: include/sequence.h [hunk 8]
```diff
@@ -3,132 +3,131 @@
...
+#define NA_BGM_SONG_OF_DOUBLE_TIME 0x4A     // Song of Double Time ("TIME_JUMP")
+#define NA_BGM_SONATA_OF_AWAKENING 0x4B     // Sonata of Awakening ("DEMO_AWAKING")
+#define NA_BGM_GORON_LULLABY 0x4C           // Goron Lullaby ("DEMO_LULLABY")
+#define NA_BGM_NEW_WAVE_BOSSA_NOVA 0x4D     // New Wave Bossa Nova ("DEMO_TIDE")
+#define NA_BGM_ELEGY_OF_EMPTINESS 0x4E      // Elegy of Emptiness ("DEMO_SHELL")
+#define NA_BGM_OATH_TO_ORDER 0x4F           // Oath to Order ("DEMO_PLEDGE")
+#define NA_BGM_SWORD_TRAINING_HALL 0x50     // Sword Training ("EXERCISE_HALL")
+#define NA_BGM_GORON_LULLABY_INTRO 0x51     // Goron Lullaby Intro ("DEMO_LULLABY_HALF")
+#define NA_BGM_LEARNED_NEW_SONG 0x52        // New Song ("OCA_YOUSEI")
+#define NA_BGM_BREMEN_MARCH 0x53            // Bremen March ("FACEMASK")
```
Comment (isghj5) [inline include/sequence.h:87]: "FACEMASK" well thats odd...

File: src/audio/code_8019AF00.c (hunk 1 above)
Comment (engineer124) [reply src/audio/code_8019AF00.c:333]: Yeah sorry, that was the exact point I was trying to make, although maybe I wasn't clear.

OoT Lost Woods -> This music plays
MM Lost Woods -> This music does not play (pre-clocktown location)
MM Woods of Mystery -> This music play (monkey puzzle)

The original name, `MAYOIMORI`, translates to `LOST WOODS`, but that name would be confusing in MM because this track does not play in MM's Lost Woods, it instead plays in the Woods of Mystery.

In trying to preserve the track name between the two games (because it is an identical track between the 2 games), I went with Saria's Song because it does match the actual ocarina Saria's Song exactly.

Plus, this song isn't only used in the Woods of Mystery. This is also the track used by the scarecrow when dancing and skipping ahead to the next half day

File: include/sequence.h (hunk 7 above)
Comment (Clownacy) [reply include/sequence.h:57]: It's not just you.

File: include/sequence.h [hunk 9]
```diff
@@ -3,132 +3,131 @@
...
-#define NA_BGM_TITLE_THEME 0x76             // Title Screen
-#define NA_BGM_DUNGEON_APPEAR 0x77          // Surfacing of Woodfall
-#define NA_BGM_WOODFALL_CLEAR 0x78          // Woodfall Clear "PURIFICATION_0"
-#define NA_BGM_SNOWHEAD_CLEAR 0x79          // Snowhead Clear "PURIFICATION_1"
-#define NA_BGM_INTO_THE_MOON 0x7B           // To the Moon
-#define NA_BGM_GOODBYE_GIANT 0x7C           //
-#define NA_BGM_TATL_AND_TAEL 0x7D           // Tatl and Tael "CHAT"
-#define NA_BGM_MOONS_DESTRUCTION 0x7E       // Moon's Destruction "MOON_VANISH"
-#define NA_BGM_NO_MUSIC 0x7F                //
-#define NA_BGM_END_CREDITS_2 0x82           // The End/Credits II "STAFFROLL2"
```
Comment (AngheloAlf) [inline include/sequence.h:131]: what happened to `NA_BGM_END_CREDITS_2`?

Comment (tom-overton) [reply include/sequence.h:131]: It never existed in MM64. It's this song that plays during the remake credits in MM3D: https://www.youtube.com/watch?v=PQxPlwcs0xw

Before removing it, I did try to play it in-game, and unsurprisingly, it just crashed. This *does* make me wonder what the final entry in the table is for `sSeqFlags` in `code_8019AF00.c`. But it certainly isn't this song, since there's no secondary staff roll in the original version of the game.

Comment (engineer124) [reply include/sequence.h:131]: The final entry in `sSeqFlags` could just be padding. Besides, the last entry is `0x7F` i.e. `NA_BGM_NO_MUSIC` so I'm not sure why `NA_BGM_END_CREDITS_2` is commented on it anyway. So that line can go and `sSeqFlags` can just be 126 or 0x7E entries long

File: include/sequence.h (hunk 7 above)
Comment (engineer124) [reply include/sequence.h:57]: I have no attachments to OCA. I only did it to keep the change minimal from OG. Feel free to change if you like

File: include/sequence.h (hunk 5 above)
Comment (tom-overton) [reply include/sequence.h:23]: "Mountain Village" is the name of the song on the official soundtrack. I've updated the comments to include the official names, but I also clarified where this plays.

File: include/sequence.h (hunk 9 above)
Comment (tom-overton) [reply include/sequence.h:131]: There was discussion on discord, but just to make sure it's all in this PR:
- Like I said above, there's no second staff roll in MM64, but...
- The credits theme in MM64 *is* split in half, so there are two sequences for it. However...
- It's at 0x7F, *not* 0x82

The code I'm about to push will update this to reflect our new understanding.

File: include/sequence.h (hunk 6 above)
Comment (tom-overton) [reply include/sequence.h:31]: I really don't see this as necessary. The sequences for the mini boss theme and for Majora's three themes are rather obviously named, so they're not hard to find. And this isn't like OoT, where there are actually two boss themes used for "normal" bosses; there's just one, and all of the four main bosses use it.

**Comment by tom-overton** [comment]
In addition to responding to reviews, the most recent commit also updates comments to include the official soundtrack's name for every sequence, and it updates a few more sequence names based on new things I've learned (e.g., some sequences do nothing but point to other sequences).

File: src/audio/code_8019AF00.c (hunk 1 above)
Comment (tom-overton) [reply src/audio/code_8019AF00.c]: Changed it back. Still not the biggest fan, but this is clearly a fight I'm not winning.

**Review by AngheloAlf — APPROVED** [review]
LGTM

File: include/sequence.h [hunk 10]
```diff
@@ -1,134 +1,138 @@
...
+#define NA_BGM_PIRATES_FORTRESS 0x14            // "Pirates' Fortress" (NA_BGM_PIRATE_AREA)
+#define NA_BGM_CLOCK_TOWN_DAY_1 0x15            // "Clock Town, First Day" - Loaded from the main Clock Town seqeuence (sequence 0x1D) (NA_BGM_DUMMY_21)
+#define NA_BGM_CLOCK_TOWN_DAY_2 0x16            // "Clock Town, Second Day" - Loaded from the main Clock Town seqeuence (sequence 0x1D) (NA_BGM_DUMMY_22)
+#define NA_BGM_CLOCK_TOWN_DAY_3 0x17            // "Clock Town, Third Day" - Loaded from the main Clock Town seqeuence (sequence 0x1D) (NA_BGM_DUMMY_23)
+#define NA_BGM_FILE_SELECT 0x18                 // File Select (NA_BGM_FILE_SELECT)
+#define NA_BGM_CLEAR_EVENT 0x19                 // "Event Clear" (NA_BGM_EVENT_CLEAR)
+#define NA_BGM_ENEMY 0x1A                       // "Battle" (NA_BGM_ENEMY)
+#define NA_BGM_BOSS 0x1B                        // "Boss Battle" (NA_BGM_BOSS00)
+#define NA_BGM_WOODFALL_TEMPLE 0x1C             // "Woodfall Temple" (NA_BGM_JUNGLE_DUNGEON)
+#define NA_BGM_CLOCK_TOWN_MAIN_SEQUENCE 0x1D    // Plays the "morning theme" at the start of a day, then loads one of sequences 0x15, 0x16, or 0x17 (NA_BGM_MARKET)
```
Comment (engineer124) [inline include/sequence.h:35]:
```suggestion
#define NA_BGM_CLOCK_TOWN_MAIN_SEQUENCE 0x1D    // Loads one of sequences 0x15, 0x16, or 0x17 (NA_BGM_MARKET). Will also play the "morning theme" if it is the start of a day
```
I think I would rephrase it this way. If you enter clocktown mid-day, this sequence will load the other sequences without playing the morning theme, it's not always played. Whether it's played or not is controlled by z_kankyo in calling either `Audio_PlayMorningSceneSequence` or `Audio_PlaySceneSequence` which will then write the correct io to this sequence

Comment (tom-overton) [reply include/sequence.h:35]: Yeah, your rephrasing makes sense. The morning theme thing is just sort of an extension to what this sequence *really* does.

**Review by engineer124 — APPROVED** [review]

File: include/sequence.h [hunk 11]
```diff
@@ -1,134 +1,138 @@
...
+#define NA_BGM_WOODFALL_CLEAR 0x78              // "Southern Swamp Clears" (NA_BGM_PURIFICATION_0)
+#define NA_BGM_SNOWHEAD_CLEAR 0x79              // "Snowhead Clear" (NA_BGM_PURIFICATION_1)
+#define NA_BGM_INTO_THE_MOON 0x7B               // "To The Moon" (NA_BGM_INTO_THE_MOON)
+#define NA_BGM_GOODBYE_GIANT 0x7C               // "The Giants' Exit" (NA_BGM_GOODBYE_GIANT)
+#define NA_BGM_TATL_AND_TAEL 0x7D               // "Tatl & Tael" (NA_BGM_CHAT)
+#define NA_BGM_MOONS_DESTRUCTION 0x7E           // "Moon's Destruction" (NA_BGM_MOON_VANISH)
+#define NA_BGM_END_CREDITS_SECOND_HALF 0x7F     // "The End / Credits" - second half. Sequence 0x74 loads this after playing the first half (NA_BGM_OCA_MUS)
+
+// This is used as a placeholder in scene commands to indicate to use the nature ambience instead.
+#define NA_BGM_NO_MUSIC 0x7F
```
Comment (hensldm) [inline include/sequence.h:135]: Interesting that it shared the same value as `NA_BGM_END_CREDITS_SECOND_HALF`.

**Review by hensldm — APPROVED** [review]
LGTM. Not sure if anyone else wants to re-review it though.
